dataflow/spanner-io/src/main/java/com/example/dataflow/SpannerReadApi.java

```diff
@@ -57,20 +67,17 @@ public interface Options extends PipelineOptions {
 
   public static void main(String[] args) {
     Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
-    Pipeline p = Pipeline.create(options);
+    Pipeline pipeline = Pipeline.create(options);
 
     String instanceId = options.getInstanceId();
     String databaseId = options.getDatabaseId();
-    // [START spanner_dataflow_readapi]
-    // Query for all the columns and rows in the specified Spanner table
-    PCollection<Struct> records = p.apply(
-        SpannerIO.read()
-            .withInstanceId(instanceId)
-            .withDatabaseId(databaseId)
-            .withTable("Singers")
-            .withColumns("singerId", "firstName", "lastName"));
-    // [END spanner_dataflow_readapi]
-
+    Dialect dialect = options.getDialect();
+    PCollection<Struct> records;
+    if (dialect == Dialect.POSTGRESQL) {
+      records = pgRead(instanceId, databaseId, pipeline);
+    } else {
+      records = spannerRead(instanceId, databaseId, pipeline);
+    }
```

Comment:
Can we call these methods `googleSqlRead` and `postgresSqlRead`?

Reply:
Done (here and elsewhere). Note that the name is `PostgreSQL` or `PostgreSql` (that is; one 's'. See also https://www.postgresql.org/)